questions.md

```diff
@@ -4,26 +4,82 @@
 
 1. On the UCI cluster, the resource request "-pe openmp 64" refers to the number of processors requested.  Does that
    request mean that your commands will use multiple processors?
+   
+   __No. 64 processors will be allocated, but your command may still only use one core__
+   
 2. In general, how do you know how many processors, how much RAM, how many files would be required/needed/written by the
    jobs you are running on the cluster?
+   
+   __You can figure out how many processors are required, RAM needed and files written by the job by prefixing the program command you are running on the cluster with ```/usr/bin/time -v```. To estimate this before running a job to get an idea of the resources you should request, you can perform this same operation on a smaller set of input data than you would run in your actual job, and then scale the resource information it provides up (i.e. use "Percent of CPU this job got", "Maximum resident set size (kbytes)", and "File system outputs" from the run on a smaller set of data to estimate processsor, RAM, and file requirements needed when the real job is run.__
+   
 3. In order to be a "good citizen", you need to have some idea of much RAM your job requires.  In particular, you need
    to know the "peak" (i.e., maximum) RAM required at any point during execution.  Show an example of the shell command
-   that you would use on a Linux machine to measure run time and peak ram usage of an arbitrary command, where the time/peak RAM values are written to a file.
+   that you would use on a Linux machine to measure run time and peak ram usage of an arbitrary command, where the time/peak RAM values    are written to a file.
+   
+   __An example of a shell command that measures and records run time and peak ram usage to file would be:__
+   
+   ```{ /usr/bin/time -v command >/dev/null;} 2>&1 | grep -E "Maximum|Elapsed" > memtime.txt``` Where "command" is the arbritrary command being run.
+   
+   __If we were to do this with a fasta file containing all _D. melanogaster_ chromosome sequences, the command and results would look like this__:
+   
+    
+    ```
+    { /usr/bin/time -v grep "A" dmel-all-chromosome-r6.24.fasta >/dev/null;} 2>&1 | grep -E "Maximum|Elapsed" > memtime.txt
+    less memtime.txt
+    Elapsed (wall clock) time (h:mm:ss or m:ss): 0:00.27
+    Maximum resident set size (kbytes): 2088
+    ```
+    
 4. What are the units of your answer for number 3?
+
+   __The units for peak RAM usage (i.e. Maximum resident set size) is kilobytes; units for run time are minutes:seconds__
+
 5. What are the bash commands for the following operations:
 
     * Checking that a file exists
+    
+    __In the directory of interest:__
+    
+    ```ls filename```
+    
```

Comment:
Actually you would want to do a conditional as it will execute if the file exists or throw/print an error to stdout or stderr. i.e.
if [ -e txt.md ]
then
    echo "Exists"
else
    echo "Does not exist"
fi
a one liner would be [ -e test.txt ] && echo "if true say something" || "if false say or do something else." here we are just doing logical operators. so && does a test to see if both are true it returns true and || is an or statement and returns something if anything is true. this is a bit confusing since it should only return if it is untrue but if you do FALSE || "something" bash will assume "something" is a TRUE value since it exists. you could just do if statements as they are easier to read and more legible. Honestly I almost exclusively use if statements and only use the one liners if i'm being lazy. to check if the file is not empty then you would replace the flag -e with -s